### PR TITLE
Basic drivetrain

### PR DESCRIPTION
**Pull request by ri-nath**

### CODE REVIEW

File: src/main/java/org/team4159/frc/robot/Constants.java [hunk 1]
```diff
@@ -1,4 +1,15 @@
 package org.team4159.frc.robot;
 
 public final class Constants {
+    public final static class CAN_ID {
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/Constants.java]: plural

File: src/main/java/org/team4159/frc/robot/Robot.java [hunk 2]
```diff
@@ -5,13 +5,11 @@
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/Robot.java:4]: can you remove all the empty methods?

File: src/main/java/org/team4159/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -1,22 +1,23 @@
 package org.team4159.frc.robot;
 
-import org.team4159.frc.robot.commands.ExampleCommand;
-import org.team4159.frc.robot.subsystems.ExampleSubsystem;
-import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj.Joystick;
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/RobotContainer.java:3]: can you separate imports by package

```
-- imports
  -- edu.wpi.first
  -- com.ctre.phoenix
  -- org.team4159.frc
```

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package org.team4159.frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java:3]: can you separate imports by package

```
-- imports
  -- edu.wpi.first
  -- com.ctre.phoenix
  -- org.team4159.frc
```

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import static org.team4159.frc.robot.Constants.*;
+
+
+public class Drivetrain extends SubsystemBase {
+    private final SpeedControllerGroup left_talons =
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java]: can you initialize these in the constructor

**Review by kaijchang — CHANGES_REQUESTED** [review]

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import static org.team4159.frc.robot.Constants.*;
+
+public class Drivetrain extends SubsystemBase {
+  private SpeedControllerGroup left_talons;
+  private SpeedControllerGroup right_talons;
+
+  public Drivetrain() {
+    left_talons = new SpeedControllerGroup(
```
Comment (kaijchang) [inline src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java:17]: make you end parenthesis consistent

**Review by kaijchang — CHANGES_REQUESTED** [review]

File: src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java (hunk 6 above)
Comment (kaijchang) [reply src/main/java/org/team4159/frc/robot/subsystems/Drivetrain.java:17]: preferably the second one where the end parenthesis is on a new line

**Review by kaijchang — APPROVED** [review]
very nice!!!!

**Comment by ri-nath** [comment]
> very nice!!!!

Thanks, U too!

**Pull request merged**